### PR TITLE
Experimental support for ExternalSource `source` in TF DALI Dataset.

### PR DESCRIPTION
**Pull request by klecki**
Signed-off-by: Krzysztof Lecki <klecki@nvidia.com>
#### Why we need this PR?
- to have feature completeness in TF compared to other frameworks.

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
     Hacking TF.
 - Affected modules and functionalities:
     *[ Describe here what was changed, added, removed. ]*
 - Key points relevant for the review:
     *[ Describe here what is the most important part that reviewers should focus on. ]*
 - Validation and testing:
     *[ Describe here if and how this PR is tested. ]*
 - Documentation (including examples):
     *[ Describe here if documentation and examples were updated. ]*

**JIRA TASK**: *[Use DALI-XXXX or NA]*

### CODE REVIEW

File: dali/python/nvidia/dali/plugin/tf.py [hunk 1]
```diff
@@ -365,30 +388,211 @@ def _get_dataset(value):
...
+                    desc = _SourceDescription(source, _SourceKind.CALLABLE,
+                                            _accepted_arg_count(source) > 0, cycle)
+                    callback = source
+            else:
+                desc = None
+                callback = None
+
+            if not iterable and cycle:
+                raise ValueError("`cycle` argument is only valid for iterable `source`")
+            return callback, desc
```
Comment (mzient) [inline dali/python/nvidia/dali/plugin/tf.py:519]: I've seen this before... can you reuse the existing function?

**Comment by klecki** [comment]
!build

**Comment by klecki** [comment]
!build

**Comment by klecki** [comment]
Replaced by a PR with new PR template.

**Pull request closed without merging**